packages/flutter_driver/lib/src/driver.dart

```diff
@@ -384,6 +385,13 @@ class FlutterDriver {
     return GetTextResult.fromJson(await _sendCommand(new GetText(finder, timeout: timeout))).text;
   }
 
+  /// Sends a string and returns a string.
+  ///
+  /// The application can respond to this by providing a handler to [enableFlutterDriverExtension].
+  Future<String> requestData(String message, { Duration timeout }) async {
```

Comment:
This is 2mm away from being a custom request handler that takes a `String` and returns a `String`. 2 things would close the 2mm gap:

1. rename this method and `DataHandler` to something that implies "you can do anything in the DataHandler and return a result"
1. have `DataHandler` return `Future<String>` instead of `String`

I feel like we should do it now so we don't have to break API or introduce an almost identical API later because users want to do more than request data.